src/app/components/cohort-charts/chart.service.js

```diff
@@ -197,8 +186,6 @@ angular.module('transmartBaseUi').factory('ChartService',
                 _.each(chartService.cs.dims, function (dim) {
                     dim.filterAll();
                 });
-                dc.filterAll();
```

Comment:
after removing this line, the filters are behaving correctly. It seems related to updateDimensions() being called extra many times that somehow "corrupts" the updating procedure.